Refactor: Remove EventsHeatMapQuery and fixing bugs

jabahamondes opened the pull request:
## Problem

There were bugs, duplicated code and no support for notebooks

## Changes

- Deleted EventsHeatMapQuery and its response types from schema and queries.
- Updated references in utils, components, and query runners to use CalendarHeatmapQuery instead.
- Adjusted related UI components and tests to reflect the removal of EventsHeatMapQuery.
- Cleaned up unused imports and code related to the EventsHeatMap functionality.> [!IMPORTANT]
- Added FF
- Added support for notebooks

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Manually

Commit:
Refactor: Remove EventsHeatMapQuery and related types

- Deleted EventsHeatMapQuery and its response types from schema and queries.
- Updated references in utils, components, and query runners to use CalendarHeatmapQuery instead.
- Adjusted related UI components and tests to reflect the removal of EventsHeatMapQuery.
- Cleaned up unused imports and code related to the EventsHeatMap functionality.

Commit:
feat(insight): Add Calendar Heatmap functionality

- Introduced Calendar Heatmap insight type in various components and logic.
- Updated UI to include Calendar Heatmap in the insights navigation and command palette.
- Implemented feature flag checks to conditionally render Calendar Heatmap options.
- Added new CalendarHeatmapQuery and related types to support the functionality.
- Enhanced filtering and summarization logic to accommodate Calendar Heatmap insights.
- Updated relevant imports and constants for consistency.

jabahamondes commented on frontend/src/scenes/insights/InsightNav/InsightsNav.tsx:
```diff
@@ -15,14 +19,20 @@ export function InsightsNav(): JSX.Element {
 
+    const { featureFlags } = useValues(featureFlagLogic)
+    const calendarHeatmapInsightEnabled = featureFlags[FEATURE_FLAGS.CALENDAR_HEATMAP_INSIGHT]
+    const filteredTabs = calendarHeatmapInsightEnabled
+        ? tabs
+        : tabs.filter((tab) => tab.type !== InsightType.CALENDAR_HEATMAP)
```
Won't happen

jabahamondes commented on frontend/src/queries/nodes/InsightViz/InsightVizDisplay.tsx:
```diff
@@ -150,7 +153,10 @@ export function InsightVizDisplay({
                 )
             case InsightType.CALENDAR_HEATMAP:
-                return <EventsHeatMapContainer context={context} />
+                if (featureFlags[FEATURE_FLAGS.CALENDAR_HEATMAP_INSIGHT]) {
+                    return <InsightCalendarHeatMapContainer context={context} />
+                }
```
Not necessary

jabahamondes commented on frontend/src/scenes/notebooks/Notebook/SlashCommands.tsx:
```diff
@@ -264,6 +266,25 @@ order by count() desc
+            chain.insertContentAt(
+                pos,
+                buildInsightVizQueryContent({
+                    kind: NodeKind.CalendarHeatmapQuery,
+                    series: [
+                        {
+                            kind: NodeKind.EventsNode,
+                            name: '$pageview',
+                            event: '$pageview',
+                        },
+                    ],
+                })
```
Not necessary

jabahamondes commented on frontend/src/scenes/notebooks/Notebook/SlashCommands.tsx:
```diff
@@ -400,7 +427,7 @@ export const SlashCommands = forwardRef<SlashCommandsRef, SlashCommandsProps>(fu
     }
     const onPressDown = (): void => {
-        setSelectedIndex(Math.min(selectedIndex + 1, SLASH_COMMANDS.length - 1))
+        setSelectedIndex(Math.min(selectedIndex + 1, slashCommands.length - 1))
```
I was consistent

jabahamondes commented on frontend/src/scenes/saved-insights/SavedInsightsFilters.tsx:
```diff
@@ -19,7 +21,15 @@ export function SavedInsightsFilters({
+    const insightTypeOptions: LemonSelectOptions<string> = calendarHeatmapInsightEnabled
+        ? INSIGHT_TYPE_OPTIONS
+        : (INSIGHT_TYPE_OPTIONS.filter(
+              (option: any) => option.value !== InsightType.CALENDAR_HEATMAP
```
Not necessary

fivemtebex commented:
Great implementation of this feature. I particularly like the approach to error handling.

jabahamondes commented on frontend/src/scenes/notebooks/Notebook/SlashCommands.tsx:
```diff
@@ -264,6 +266,25 @@ order by count() desc
+    {
+        title: 'Calendar Heatmap',
+        search: 'calendar heatmap insight',
+        icon: <IconHogQL />,
```
For now I will leave it

jabahamondes commented on frontend/src/scenes/saved-insights/newInsightsMenu.tsx:
```diff
@@ -1,35 +1,48 @@
+    const calendarHeatmapInsightEnabled = featureFlags[FEATURE_FLAGS.CALENDAR_HEATMAP_INSIGHT]
+
     const menuEntries = Object.entries(INSIGHT_TYPES_METADATA).filter(
-        ([insightType]) => insightType !== InsightType.JSON
+        ([insightType]) =>
+            insightType !== InsightType.JSON &&
+            (calendarHeatmapInsightEnabled || insightType !== InsightType.CALENDAR_HEATMAP)
     )
```
Not doing it bc I'll revert this in the future

jabahamondes commented on frontend/src/scenes/web-analytics/CalendarHeatMap/CalendarHeatMapCell.tsx:
```diff
@@ -40,7 +40,7 @@ export function getBackgroundAndTextColor({
     let backgroundColorSaturation =
         values.maxValue === 0 ? 0 : Math.max(Math.min(0.8, values.value / values.maxValue), 0.3)
```
It's expected

jabahamondes commented on frontend/src/scenes/web-analytics/WebActiveHoursHeatmap/WebActiveHoursHeatmap.tsx:
```diff
@@ -27,24 +28,24 @@ import {
     context?: QueryContext
     cachedResults?: AnyResponseType
 }
+let uniqueNode = 0
```
Copied from the codebase

jabahamondes commented on frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:
```diff
@@ -1890,7 +1893,49 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
+                                              </>
+                                          ),
+                                      },
+                                      insightProps: createInsightProps(TileId.ACTIVE_HOURS, ActiveHoursTab.UNIQUE),
```
Good catch

jabahamondes commented on frontend/src/scenes/saved-insights/newInsightsMenu.tsx:
```diff
@@ -1,35 +1,48 @@
-export function overlayForNewInsightMenu(dataAttr: string): ReactNode[] {
+export function OverlayForNewInsightMenu({ dataAttr }: { dataAttr: string }): JSX.Element {
+    const { featureFlags } = useValues(featureFlagLogic)
+    const calendarHeatmapInsightEnabled = featureFlags[FEATURE_FLAGS.CALENDAR_HEATMAP_INSIGHT]
```
Not doing it bc I'll revert it in the future

Commit:
PRs comments

Commit:
reverting mistake

Commit:
Removing code that is automaticalled removed

Commit:
Refactor: Clean up CalendarHeatmapQueryRunner and related imports

- Removed unused function  from CalendarHeatmapQueryRunner.
- Updated  method to handle empty properties more gracefully.
- Removed commented-out code and unnecessary  import from web analytics query runner.
- Cleaned up code for better readability and maintainability.

Commit:
fix(filters): Improve type safety for insight type options

- Updated filtering logic in SavedInsightsFilters to ensure type safety by using type guards.
- Added missing LemonSelectOption import for better clarity and consistency in the component.
